### DISCUSSION TITLE
CKAN Stuck on old directory

### DISCUSSION THREAD

**Discussion by beny873** [KSP-CKAN/CKAN#3290] [Q&A] [answered]
Afternoon everyone!

So I have a peculiar problem and I'm stumped on how to fix it. 

Windows 10

The short of it is this.

```
Unhandled Exception: System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\SteamLibrary\SteamApps\common\Kerbal Space Program\Mods\Mods Cache'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.Directory.InternalCreateDirectory(String fullPath, String path, Object dirSecurityObj, Boolean checkHost)
   at System.IO.Directory.InternalCreateDirectoryHelper(String path, Boolean checkHost)
   at CKAN.KSPManager.LoadInstancesFromRegistry()
   at CKAN.CmdLine.KSP.<>c__DisplayClass9_0.<RunSubCommand>b__0(String option, Object suboptions)
   at CommandLine.Parser.ParseArgumentsStrict(String[] args, Object options, Action`2 onVerbCommand, Action onFail)
   at CKAN.CmdLine.KSP.RunSubCommand(KSPManager manager, CommonOptions opts, SubCommandOptions unparsed)
   at CKAN.CmdLine.MainClass.Execute(KSPManager manager, CommonOptions opts, String[] args)
   at CKAN.CmdLine.MainClass.Main(String[] args)
```
   

Seems pretty straight forward. I no longer have a D: Drive. Except my admittedly amateur level coding cannot solve it. 
   
Context: My last KSP play through was over a year ago on 1.8 with JNSQ, Principia, etc etc. Since then my PC has undergone a major upgrade, only the storage devices remain of the old components. I completely uninstalled KSP, deleted everything in the folder, the folder itself, CKAN, everything. 

I recently felt my inner rocketnerd calling to me again and wanted to jump back into 1.11. Installed KSP from Steam, did a quick run as a test and it was fine. Downloaded the CKAN.exe like before, ran it and I was greeted with the error above in a flash of CMD prompt before it closes. This happens with the .exe in the KSP root directory and on the desktop. 

Thing is, this here is my current KSP path now:
G:\Games\SteamLibrary\steamapps\common\Kerbal Space Program

It's on a new m.2 SSD, it's a completely fresh game. 

Started the usual motions of running as admin, deleting and reinstalling, redownloading CKAN, etc etc. No dice. The usual method of backing up mods and deleting the CKAN folder don't apply here since there is no CKAN folder and no mods to speak of. 

Went into more detail and started trying to fix it via powershell, still no dice. Every command I try to execute returns the same error. 
```
"PS G:\Games\SteamLibrary\steamapps\common\Kerbal Space Program> .\ckan.exe ksp list" = as above
"PS G:\Games\SteamLibrary\steamapps\common\Kerbal Space Program> .\ckan.exe cache list" = as above

```
I even went into regedit, found the CKAN entry, edited it with a new path, still no dice. I then deleted the registry entry completely, still no dice. 

Trying to change the file path via powershell with the 'ksp add' command results in the same error. 

I'm stumped lads. Any help would be greatly appreciated.

**Comment by HebaruSan** [comment, marked as answer]
Deleting this file should allow you to start over:

- `C:\Users\<Your user name>\AppData\Local\CKAN\config.json`

That's where the (formerly) registry settings live now, essentially because the registry is a platform-specific notion.

**Reply by beny873** [reply]
omg -_-

I did this once. Went through all of appdata, Roaming, LocalLow, all of it. 

It was still doing. 

Tried doing it again....BAM, works like a charm. Trigger me timbers. 

Thanks Hebaru :D
